app/uk/gov/hmrc/agentclientrelationships/connectors/DesConnector.scala

```diff
@@ -53,25 +56,30 @@ object ClientRelationship {
 class DesConnector @Inject()(@Named("des-baseUrl") baseUrl: URL,
                              @Named("des.authorizationToken") authorizationToken: String,
                              @Named("des.environment") environment: String,
-                             httpGet: HttpGet) {
+                             httpGet: HttpGet,
+                             metrics: Metrics)
+  extends HttpAPIMonitor {
+  override val kenshooRegistry: MetricRegistry = metrics.defaultRegistry
 
   def getNinoFor(mtdbsa: MtdItId)(implicit hc: HeaderCarrier): Future[Nino] = {
     val url = new URL(baseUrl, s"/registration/business-details/mtdbsa/${ encodePathSegment(mtdbsa.value) }")
-    getWithDesHeaders[BusinessDetails](url).map(_.nino)
+    getWithDesHeaders[BusinessDetails]("GetRegistrationBusinessDetailsByMtdbsa", url).map(_.nino)
   }
 
   def getClientSaAgentSaReferences(nino: Nino)(implicit hc: HeaderCarrier): Future[Seq[SaAgentReference]] = {
     val url = new URL(baseUrl, s"/registration/relationship/nino/${ encodePathSegment(nino.value) }")
-    getWithDesHeaders[ClientRelationship](url).map(_.agents.flatMap {
+    getWithDesHeaders[ClientRelationship]("GetStatusAgentRelationship", url).map(_.agents.flatMap {
       case Agent(true, Some(agentId), None) => Some(agentId)
       case _ => None
     })
   }
 
-  private def getWithDesHeaders[A: HttpReads](url: URL)(implicit hc: HeaderCarrier): Future[A] = {
+  private def getWithDesHeaders[A: HttpReads](apiName: String, url: URL)(implicit hc: HeaderCarrier): Future[A] = {
     val desHeaderCarrier = hc.copy(
       authorization = Some(Authorization(s"Bearer $authorizationToken")),
       extraHeaders = hc.extraHeaders :+ "Environment" -> environment)
-    httpGet.GET[A](url.toString)(implicitly[HttpReads[A]], desHeaderCarrier)
+    monitor(s"ConsumedAPI-DES-$apiName-GET") {
```

Comment:
I can't find a test for this (and it would be good to have one) - I think examples of testing this kind of monitoring can be found in some of our other services.